### PR TITLE
Rewrite merge_cluster_roles script in Go

### PR DESCRIPTION
**Pull request by dgrisonnet**
* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: manifests/0000_50_cluster-monitoring-operator_02-role.yaml [hunk 1]
```diff
@@ -30,31 +30,77 @@
 apiVersion: rbac.authorization.k8s.io/v1beta1
 kind: ClusterRole
 metadata:
+  creationTimestamp: null
```
Comment (dgrisonnet) [inline manifests/0000_50_cluster-monitoring-operator_02-role.yaml:33]: This is the only addition to the generated manifest. The other changes are only positional ones.

This addition is not intended but I can't get rid of it for some reason.

**Comment by paulfantom** [comment]
Awesome!!!

One nit: could you remove the mention of python from https://github.com/openshift/cluster-monitoring-operator/blob/master/CONTRIBUTING.md#required-tools? It was needed only for this script.

File: manifests/0000_50_cluster-monitoring-operator_02-role.yaml (hunk 1 above)
Comment (lilic) [reply manifests/0000_50_cluster-monitoring-operator_02-role.yaml:33]: We seem to have that elsewhere, so it's okay. https://github.com/openshift/cluster-monitoring-operator/blob/d32a7b0d759d28221b4a4a3f5429c8b00738790a/manifests/0000_50_cluster-monitoring-operator_00_0servicemonitor-custom-resource-definition.yaml#L6

**Comment by dgrisonnet** [comment]
@paulfantom while I was at it, I also removed `python-yaml` from the cmo-tooling image.

**Comment by s-urbaniak** [comment]
wonderful, thank you! :tada:

**Comment by paulfantom** [comment]
@dgrisonnet could you rebase?

/unhold

**Pull request merged**